8289770: Remove Windows version macro from ShellFolder2.cpp

aivanov-jdk opened the pull request:
This clean-up PR removes unused Windows version macro from `ShellFolder2.cpp`.

`IS_WINVISTA` was not used at all.

`IS_WINXP` guarded support for icons with alpha channel. It is now safe to assume Java runs on a Windows version later than Windows XP. Java launchers specify 6.0 as the minimum OS version which corresponds to Windows Vista.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8289770](https://bugs.openjdk.org/browse/JDK-8289770): Remove Windows version macro from ShellFolder2.cpp (**Bug** - P4)


### Reviewers
 * [Julian Waters](https://openjdk.org/census#jwaters) (@TheShermanTanker - Committer)
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Committer)
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18736/head:pull/18736` \
`$ git checkout pull/18736`

Update a local copy of the PR: \
`$ git checkout pull/18736` \
`$ git pull https://git.openjdk.org/jdk.git pull/18736/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18736`

View PR using the GUI difftool: \
`$ git pr show -t 18736`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18736.diff">https://git.openjdk.org/jdk/pull/18736.diff</a>

</details>


### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18736#issuecomment-2049309266)

TejeshR13 reviewed (commented):
Can you update JBS title and description regarding "support for icons with alpha channel" too. These updates are visible only in PR and not in JBS.

aivanov-jdk commented:
> Can you update JBS title and description regarding "support for icons with alpha channel" too. These updates are visible only in PR and not in JBS.

Updated the JBS subject.

Added [a comment](https://bugs.openjdk.org/browse/JDK-8289770?focusedId=14664984&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14664984) with break-up of macro usages.

mrserb commented on src/java.desktop/windows/native/libawt/windows/ShellFolder2.cpp:
```diff
@@ -119,17 +119,6 @@ static jstring lsSize;
-#endif
-#ifndef IS_WINVISTA
-#define IS_WINVISTA (!(::GetVersion() & 0x80000000) && LOBYTE(LOWORD(::GetVersion())) >= 6)
-#endif
```
there are exactly the same macro in awt.h which are mostly unused except "IS_WIN8"

aivanov-jdk replied:
There's a separate bug for removing the macro from `awt.h`:

[JDK-8289772](https://bugs.openjdk.org/browse/JDK-8289772): Remove obsolete Windows version macros: IS_WIN2000, IS_WINXP, IS_WINVISTA

Why am I taking `ShellFolder2.cpp` separately? The comment in the `ShellFolder2.cpp` file suggest these are copies from `awt.h`. This makes `ShellFolder2.cpp` isolated from any other usage. Removing the macro cleans up a single file, easy to do, easy to review.

Removing the macro from `awt.h` is not as simple, some macro are used in the code. For example, `IS_WIN8` is used in `awt_Toolkit.cpp` to determine whether touch keyboard handling is enabled.

https://github.com/openjdk/jdk/blob/58911ccc2c48b4b5dd2ebc9d2a44dcf3737eca50/src/java.desktop/windows/native/libawt/windows/awt_Toolkit.cpp#L669

Then `IS_WINVISTA` is used in the code which determines whether DWM is available; `IS_WINXP` is used in many places to determine menu bar colors, to handle subclassing default Windows controls. Removing the macro will make the code cleaner but it requires modifying the code. Eventually, the task may be broken into smaller pieces.

The usage of `IS_WIN8` could be replaced with [`IsWindows8OrGreater`](https://learn.microsoft.com/en-us/windows/win32/api/VersionHelpers/nf-versionhelpers-iswindows8orgreater) which is part of [Version Helper functions](https://learn.microsoft.com/en-us/windows/win32/sysinfo/version-helper-apis) and is the recommended method to determine the version of the Windows OS.
